test-case/check-suspend-resume.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+##
+## Case Name: check suspend/resume status
+## Preconditions:
+##    N/A
+## Description:
+##    Run the suspend/resume command to check device status
+## Case step:
+##    1. switch suspend/resume operation
+##    2. use rtcwake -m mem command to do suspend/resume
+##    3. check command return value
+##    4. check dmesg errors
+##    5. check wakeup increase
+## Expect result:
+##    suspend/resume recover
+##    check kernel log and find no errors
+##
+
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+random_min=3    # wait time should >= 3 for other device wakeup from sleep
+random_max=20
+
+OPT_OPT_lst['l']='loop'     OPT_DESC_lst['l']='loop count'
+OPT_PARM_lst['l']=1         OPT_VALUE_lst['l']=5
+
+OPT_OPT_lst['p']='power'    OPT_DESC_lst['p']="suspend/resume type from /sys/power/mem_sleep"
+OPT_PARM_lst['p']=1         OPT_VALUE_lst['p']=""
+
+OPT_OPT_lst['s']='sleep'    OPT_DESC_lst['s']='suspend/resume command:rtcwake sleep option'
+OPT_PARM_lst['s']=1         OPT_VALUE_lst['s']=5
+
+OPT_OPT_lst['w']='wait'     OPT_DESC_lst['w']='wait time after suspend/resume wakeup'
+OPT_PARM_lst['w']=1         OPT_VALUE_lst['w']=5
+
+OPT_OPT_lst['r']='random'   OPT_DESC_lst['r']="Random setup wait/sleep time, time range is [$random_min-$random_max], this option will overwait s & w option"
+OPT_PARM_lst['r']=0         OPT_VALUE_lst['r']=0
+
+func_opt_parse_option $*
+func_lib_check_sudo
+func_lib_setup_kernel_last_line
+
+power=${OPT_VALUE_lst['p']}
+# switch type
+if [ "$power" ]; then
```

Comment:
If user doesn't set the power mode, the susprend/resume mode will be random. Is this what we expected?

Reply:
it is not the random, just use the system current setting for it  
This value will be dump by the test-case to run  
Current I am wait for confirm the default value of `power`

Reply:
Yes, this is what I mean. It depends on the current setting. But the current setting maybe s2idle or deep or standby. If we have a default value for power mode, it can fix this issue.

Reply:
@libinyang @Bin-QA It would not be random. The system will always have default value. We should test with the system default value.
We may get into some trouble if we try to test in other modes.

Reply:
But the system's default value will be different on different platforms based on my understanding. So without appointing the mode, we can't tell we are doing S3 or s2idle for the case. What's more, someone may temporarily uses the DUT to test suspend/resume in another mode.

Reply:
rename this parameter from 'power' to 'type'